packages/web-components/src/component-mixins/cta/cta.ts

```diff
@@ -325,26 +325,31 @@ const CTAMixin = <T extends Constructor<HTMLElement>>(Base: T) => {
           (videoName === null || videoName === 'null')) ||
         changedProperties.has('videoDescription')
       ) {
-        // Wait for the container to be ready without blocking.
-        customElements
-          .whenDefined(`${c4dPrefix}-video-cta-container`)
-          .then(() => {
-            this.dispatchEvent(
-              new CustomEvent(eventRequestVideoData, {
-                bubbles: true,
-                cancelable: true,
-                composed: true,
-                detail: {
-                  videoName,
-                  videoDescription,
-                  href,
-                },
-              })
-            );
-          });
+        if (typeof videoDuration === 'undefined') {
+          // Wait for the container to be ready without blocking.
+          customElements
+            .whenDefined(`${c4dPrefix}-video-cta-container`)
+            .then(() => {
+              this.dispatchEvent(
+                new CustomEvent(eventRequestVideoData, {
+                  bubbles: true,
+                  cancelable: true,
+                  composed: true,
+                  detail: {
+                    videoName,
+                    videoDescription,
+                    href,
+                  },
+                })
+              );
+            });
+        }
       }
 
-      if (ctaType === CTA_TYPE.VIDEO && this.offsetWidth > 0) {
+      if (
+        ctaType === CTA_TYPE.VIDEO &&
+        this?.shadowRoot?.querySelector(':first-child')?.offsetWidth > 0
```

Comment:
Ended up using a simpler selector so we're not hardcoding card-specific things. I also checked for other CTAMixin uses, and of those I couldn't find any that both used a `videoThumbnailUrl` and were not `C4DCard` children. Those are:

* C4DButton
* C4DButtonGroupItem
* C4DButtonCTA
* C4DLinkWithIcon
* C4DCardFooter
* C4DFeatureCardFooter
* C4DFeatureCTAFooter
* C4DTextCTA

The other way to look at is to see what is actually reading from `videoThumbnailUrl`. Those places are:

* [card.ts](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/card/card.ts#L163)
* [card-in-card.ts](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/card-in-card//card-in-card.ts#L31)
* [card-cta.ts](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/cta/card-cta.ts#L74)
* [feature-cta.ts](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/cta/feature-cta.ts#L74)

All of those are descendants of `C4DCard`. Does give a good list to test from.